The type declaration order influences the parsing

The following program `sample.go` triggers a panic:

```go
package main

import "fmt"

type Hello struct {
	Goodbye GoodbyeProvider
}

func main() {
	a := &Hello{}

	fmt.Println(a)
}

type GoodbyeProvider func(message string) string
```

Expected result:
```console
$ go run ./sample.go
&{<nil>}
```

Got:
```console
$ yaegi ./sample.go
panic: reflect.StructOf: field 0 has no type

goroutine 1 [running]:
reflect.StructOf(0xc0001cd3b0, 0x1, 0x1, 0x0, 0x0)
        /home/ldez/.gvm/gos/go1.12.5/src/reflect/type.go:2369 +0x357c
github.com/containous/yaegi/interp.(*Type).TypeOf(0xc0001d05b0, 0x5ee845, 0xbf2b20)
        /home/ldez/sources/go/src/github.com/containous/yaegi/interp/type.go:609 +0x722
github.com/containous/yaegi/interp.(*Scope).add(0xc000158640, 0xc0001d05b0, 0x5d300b)
        /home/ldez/sources/go/src/github.com/containous/yaegi/interp/scope.go:161 +0x1d1
github.com/containous/yaegi/interp.(*Interpreter).Cfg.func2(0xc0001dd600)
        /home/ldez/sources/go/src/github.com/containous/yaegi/interp/cfg.go:716 +0x9293
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001dd600, 0xc0001b3d50, 0xc0001b3d88)
        /home/ldez/sources/go/src/github.com/containous/yaegi/interp/interp.go:106 +0xaf
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001dd500, 0xc0001b3d50, 0xc0001b3d88)
        /home/ldez/sources/go/src/github.com/containous/yaegi/interp/interp.go:103 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001dd300, 0xc0001b3d50, 0xc0001b3d88)
        /home/ldez/sources/go/src/github.com/containous/yaegi/interp/interp.go:103 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001dd200, 0xc0001b3d50, 0xc0001b3d88)
        /home/ldez/sources/go/src/github.com/containous/yaegi/interp/interp.go:103 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001dcd00, 0xc0001b3d50, 0xc0001b3d88)
        /home/ldez/sources/go/src/github.com/containous/yaegi/interp/interp.go:103 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001dc000, 0xc0001b3d50, 0xc0001b3d88)
        /home/ldez/sources/go/src/github.com/containous/yaegi/interp/interp.go:103 +0x70
github.com/containous/yaegi/interp.(*Interpreter).Cfg(0xc00015a0c0, 0xc0001dc000, 0xc00016a080, 0x4, 0x0, 0x0, 0x4)The 
        /home/ldez/sources/go/src/github.com/containous/yaegi/interp/cfg.go:24 +0x1b2
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc00015a0c0, 0xc0001d4000, 0xae, 0x2ae, 0xc0001d4000, 0xae, 0x0, 0xc0001b48f0)
        /home/ldez/sources/go/src/github.com/containous/yaegi/interp/interp.go:242 +0x135
main.main()
        /home/ldez/sources/go/src/github.com/containous/yaegi/yaegi.go:59 +0x3ec
exit status 2

```

Works if:
```go
package main

import "fmt"

type GoodbyeProvider func(message string) string

type Hello struct {
	Goodbye GoodbyeProvider
}

func main() {
	a := &Hello{}

	fmt.Println(a)
}

```

--- a/_test/type12.go
+++ b/_test/type12.go
@@ -1,18 +1,27 @@
 package main
 
-import "fmt"
-
-type Hello struct {
-	Goodbye GoodbyeProvider
+type T1 struct {
+	T2 *T2
 }
+
+func (t *T1) Get() string {
+	return t.T2.V().Name
+}
+
+type T2 struct {
+	Name string
+}
+
+func (t *T2) V() *T2 {
+	if t == nil {
+		return defaultT2
+	}
+	return t
+}
+
+var defaultT2 = &T2{"no name"}
 
 func main() {
-	a := &Hello{}
-
-	fmt.Println(a)
+	t := &T1{}
+	println(t.Get())
 }
-
-type GoodbyeProvider func(message string) string
-
-// Output:
-// &{<nil>}

--- a/interp/type.go
+++ b/interp/type.go
@@ -300,6 +300,12 @@ func nodeType(interp *Interpreter, scope *Scope, n *Node) (*Type, error) {
 	case Ident:
 		if sym, _, found := scope.lookup(n.ident); found {
 			t = sym.typ
+			if t.incomplete && t.node != n {
+				m := t.method
+				t, err = nodeType(interp, scope, t.node)
+				t.method = m
+				sym.typ = t
+			}
 		} else {
 			t.incomplete = true
 			scope.sym[n.ident] = &Symbol{kind: Typ, typ: t}
@@ -445,7 +451,9 @@ func init() {
 func (t *Type) finalize() (*Type, error) {
 	var err CfgError
 	if t.incomplete {
+		m := t.method
 		t, err = nodeType(t.node.interp, t.scope, t.node)
+		t.method = m
 		t.node.typ = t
 		if t.incomplete && err == nil {
 			err = t.node.cfgError("incomplete type")

